library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -258,11 +258,17 @@ class Client() {
         ffiClient.applySignatureRequest(signatureRequest)
     }
 
-    suspend fun addAccount(newAccount: SigningKey) {
-        val signatureRequest =
-            ffiClient.addWallet(newAccount.address.lowercase())
-        handleSignature(signatureRequest, newAccount)
-        ffiClient.applySignatureRequest(signatureRequest)
+    @DelicateApi("This function is delicate and should be used with caution. Adding a wallet already associated with an inboxId will cause the wallet to loose access to that inbox. See: inboxIdFromAddress(address)")
+    suspend fun addAccount(newAccount: SigningKey, changeInboxId: Boolean = false) {
```

Comment:
What about something with "allow" in it? It will only change the inbox ID if the new account already has one, which is not going to be true most of the time.

Something like `allowReassignInboxId`?